src/psyclone/generator.py

```diff
@@ -242,33 +247,61 @@ def generate(filename, api="", kernel_paths=None, script_name=None,
     ast, invoke_info = parse(filename, api=api, invoke_name="invoke",
                              kernel_paths=kernel_paths,
                              line_length=line_length)
-    if api != "gocean1.0":
+
+    # TODO issue #1618 remove temporary flag, associated logic and Alg class
+    # plus tests (and ast variable).
+    #
+    # Temporary flag to allow optional testing of new LFRic metadata
+    # implementation (mainly that the PSyIR works with algorithm-layer
+    # code) whilst keeping the original implementation as default
+    # until it is working.
+    lfric_testing = False
+
+    if api in API_WITHOUT_ALGORITHM or \
+       (api == "dynamo0.3" and not lfric_testing):
         psy = PSyFactory(api, distributed_memory=distributed_memory)\
             .create(invoke_info)
         if script_name is not None:
             handle_script(script_name, psy, "trans")
+        alg_gen = None
 
-    alg_gen = None
-
-    if api == "gocean1.0":
+    elif api == "gocean1.0" or (api == "dynamo0.3" and lfric_testing):
         # Create language-level PSyIR from the Algorithm file
         reader = FortranReader()
-        psyir = reader.psyir_from_file(filename)
+        if api == "dynamo0.3":
+            # avoid undeclared builtin errors in PSyIR by adding "use
```

Comment:
Please bracket with TODO or something to show that this is going to be removed.

Reply:
Added todo.